frontend/public/kubevirt/components/modals/delete-device-modal.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as _ from 'lodash';
+
+import { PromiseComponent } from '../utils/okdutils';
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory/okdfactory';
+import { k8sPatch } from '../../module/okdk8s';
+import { VirtualMachineModel } from '../../models';
+import { NETWORK, DISK } from '../utils/constants';
+
+class DeleteDeviceModal extends PromiseComponent {
+  constructor(props) {
+    super(props);
+    this._cancel = this.props.cancel.bind(this);
+    this._submit = this.submit.bind(this);
+  }
+
+  submit(event) {
+    event.preventDefault();
+    const { vm, type, device } = this.props;
+
+    const deviceType = type === NETWORK ? {type: 'interfaces', spec: 'networks'} : { type: 'disks', spec: 'volumes'};
+    const devices = vm.spec.template.spec.domain.devices[deviceType.type];
+
+    const deviceIndex = devices.findIndex(d => d.name === device.name);
+    const specIndex = vm.spec.template.spec[deviceType.spec].findIndex(spec => spec.name === device.name || spec.name === device.volumeName);
+
+    const patch = [
+      {
+        op: 'remove',
+        path: `/spec/template/spec/domain/devices/${deviceType.type}/${deviceIndex}`,
+      },
+      {
+        op: 'remove',
+        path: `/spec/template/spec/${deviceType.spec}/${specIndex}`,
+      }
+    ];
+
+    // disk may have dataVolumeTemplate defined that should be deleted too
+    if (type === DISK && device.volume.dataVolume && vm.spec.dataVolumeTemplates) {
+      const dataVolumeIndex = vm.spec.dataVolumeTemplates.findIndex(dataVolume => _.get(dataVolume, 'metadata.name') === device.volume.dataVolume.name);
+      if (dataVolumeIndex !== -1) {
+        patch.push({
+          op: 'remove',
+          path: `/spec/dataVolumeTemplates/${dataVolumeIndex}`,
+        });
+      }
+    }
+
+    // if pod network is deleted, we need to set autoAttachPodInterface to false
+    if (type === NETWORK && device.network.pod) {
+      const op = _.has(vm, 'spec.domain.devices.autoAttachPodInterface') ? 'replace' : 'add';
+      patch.push({
+        op,
+        path: '/spec/template/spec/domain/devices/autoAttachPodInterface',
+        value: false
+      });
+    }
+
+    const promise = k8sPatch(VirtualMachineModel, vm, patch);
+    this.handlePromise(promise).then(this.props.close);
+  }
+
+  render () {
+    const {vm, device} = this.props;
+    return <form onSubmit={this._submit} name="form">
```

Comment:
any reason for not using pf-react's `Form`?

Reply:
switched to pf-react `Form`